### PR TITLE
Fix loading license report templates from config bundle

### PR DESCRIPTION
**Pull request by pfistererm**
So far, license report templates packagd into a configuration bundle were not loaded correctly (only loading from file system was working). This PR fixes this by moving the loading of templates to ConfigFileHandler.

## Type of request

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactoring 
- [ ] Documentation or documentation changes

#### Initiator
Matthias Pfisterer

## Checklist:

- [x] All the commits are signed.
- [x] My code follows the code style of this project.
- [x] I agree with die CLA.
- [x] I have read the CONTRIBUTING docs.
- [x] I have added/updated necessary documentation (if appropriate).

### CODE REVIEW

File: licensescout-core/src/main/java/org/aposin/licensescout/configuration/FilesystemConfigFileHandler.java [hunk 1]
```diff
@@ -196,4 +196,36 @@ public InputStream getFilteredVendorNamesInputStream() throws IOException {
...
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean hasTemplateFile(String filename) throws IOException {
+        if (filename != null) {
+            final File templateFile = new File(filename);
+            return (templateFile.isFile() && templateFile.canRead());
+        } else {
+            return false;
+        }
```
Comment (d-gs) [inline licensescout-core/src/main/java/org/aposin/licensescout/configuration/FilesystemConfigFileHandler.java:210]: Simplify this else

```suggestion
        }
        return false;
```

File: licensescout-core/src/main/java/org/aposin/licensescout/configuration/FilesystemConfigFileHandler.java [hunk 2]
```diff
@@ -196,4 +196,36 @@ public InputStream getFilteredVendorNamesInputStream() throws IOException {
...
+            final File templateFile = new File(filename);
+            if (templateFile.exists()) {
+                return templateFile;
+            } else {
+                return null;
+            }
+        } else {
+            return null;
+        }
+
```
Comment (d-gs) [inline licensescout-core/src/main/java/org/aposin/licensescout/configuration/FilesystemConfigFileHandler.java:228]: Simplify this else chain:

```suggestion
            }
        }
        return null;
```

File: licensescout-core/src/main/java/org/aposin/licensescout/configuration/ZipConfigFileHandler.java [hunk 3]
```diff
@@ -48,10 +49,25 @@ public ZipConfigFileHandler(File artifactFile, ConfigFileParameters configFilePa
...
-            byte[] buffer = IOUtils.toByteArray(zipInputStream);
-            return new ByteArrayInputStream(buffer);
+            if (entry != null) {
+                InputStream zipInputStream = zipFile.getInputStream(entry);
+                //Read the zip input stream fully into memory
+                byte[] buffer = IOUtils.toByteArray(zipInputStream);
+                return new ByteArrayInputStream(buffer);
+            } else {
+                return null;
+            }
```
Comment (d-gs) [inline licensescout-core/src/main/java/org/aposin/licensescout/configuration/ZipConfigFileHandler.java:59]: Simplify else:

```suggestion
            }
            return null;
```

File: licensescout-core/src/main/java/org/aposin/licensescout/configuration/ZipConfigFileHandler.java [hunk 4]
```diff
@@ -48,10 +49,25 @@ public ZipConfigFileHandler(File artifactFile, ConfigFileParameters configFilePa
...
+
+    private boolean hasEntry(final String entryName) throws IOException {
+        if (entryName != null) {
+            try (ZipFile zipFile = new ZipFile(artifactFile)) {
+                ZipEntry entry = zipFile.getEntry(entryName);
+                return entry != null;
+            }
+        } else {
+            return false;
         }
```
Comment (d-gs) [inline licensescout-core/src/main/java/org/aposin/licensescout/configuration/ZipConfigFileHandler.java:71]: Simplify:

```suggestion
        }
        return false;
```

File: licensescout-core/src/main/java/org/aposin/licensescout/configuration/ZipConfigFileHandler.java [hunk 5]
```diff
@@ -118,4 +134,32 @@ public InputStream getCheckedArchivesInputStream() throws IOException {
...
+                FileUtils.copyInputStreamToFile(is, tmpFile);
+                return tmpFile;
+            } else {
+                return null;
+            }
+        } else {
+            return null;
+        }
+    }
+
```
Comment (d-gs) [inline licensescout-core/src/main/java/org/aposin/licensescout/configuration/ZipConfigFileHandler.java:164]: Simplify:

```suggestion
            }
            return null;
        }
        return null;
    }
```

File: licensescout-core/src/main/java/org/aposin/licensescout/configuration/FilesystemConfigFileHandler.java (hunk 1 above)
Comment (pfistererm) [reply licensescout-core/src/main/java/org/aposin/licensescout/configuration/FilesystemConfigFileHandler.java]: Changed as requested.

File: licensescout-core/src/main/java/org/aposin/licensescout/configuration/FilesystemConfigFileHandler.java (hunk 2 above)
Comment (pfistererm) [reply licensescout-core/src/main/java/org/aposin/licensescout/configuration/FilesystemConfigFileHandler.java]: Changed as requested.

File: licensescout-core/src/main/java/org/aposin/licensescout/configuration/ZipConfigFileHandler.java (hunk 3 above)
Comment (pfistererm) [reply licensescout-core/src/main/java/org/aposin/licensescout/configuration/ZipConfigFileHandler.java]: Changed as requested.

File: licensescout-core/src/main/java/org/aposin/licensescout/configuration/ZipConfigFileHandler.java (hunk 4 above)
Comment (pfistererm) [reply licensescout-core/src/main/java/org/aposin/licensescout/configuration/ZipConfigFileHandler.java]: Changed as requested.

File: licensescout-core/src/main/java/org/aposin/licensescout/configuration/ZipConfigFileHandler.java (hunk 5 above)
Comment (pfistererm) [reply licensescout-core/src/main/java/org/aposin/licensescout/configuration/ZipConfigFileHandler.java]: Changed as requested.

**Review by d-gs — APPROVED** [review]

**Pull request merged**